### ISSUE TITLE
:fire: NativeError not converted to Error in PhoneAuthListener

### ISSUE THREAD

**Issue by mikehardy** [invertase/react-native-firebase#2041]
## Issue

TL;DR - I think there is a type error in PhoneAuthSnapshot.error and I have a fix locally that seems to work, but I may be misunderstanding NativeError->JS conversion

Context: 

I was unable to access the phoneAuthSnapshot.error.code even though it was populated when I entered (e.g.) invalid phone numbers or sending real ones on an emulator while testing phone verification on Android, this is what happens when you are:

1.  in the state_changed listener on verifyPhoneNumber and you
1.  hit phoneAuthSnapshot.state == firebase.auth.PhoneAuthState.ERROR and then you  
1.  `console.log(phoneAuthSnapshot)` 

```
04-08 16:49:37.754 28307 31219 I ReactNativeJS: 'verification error, phoneAuthSnapshot: ', { verificationId: undefined,
04-08 16:49:37.754 28307 31219 I ReactNativeJS:   code: null,
04-08 16:49:37.754 28307 31219 I ReactNativeJS:   error: 
04-08 16:49:37.754 28307 31219 I ReactNativeJS:    { [Error: This app is not authorized to use Firebase Authentication. Please verifythat the correct package name and SHA-1 are configured in the Firebase Console. [ App validation failed. Is app running on a physical device? ]]
04-08 16:49:37.754 28307 31219 I ReactNativeJS:      code: 'auth/app-not-authorized',
04-08 16:49:37.754 28307 31219 I ReactNativeJS:      nativeErrorMessage: 'This app is not authorized to use Firebase Authentication. Please verifythat the correct package name and SHA-1 are configured in the Firebase Console. [ App validation failed. Is app running on a physical device? ]',
04-08 16:49:37.754 28307 31219 I ReactNativeJS:      line: 119092,
04-08 16:49:37.754 28307 31219 I ReactNativeJS:      column: 26,
04-08 16:49:37.754 28307 31219 I ReactNativeJS:      sourceURL: 'http://10.0.2.2:8081/index.android.delta?platform=android&dev=true&minify=false' },
04-08 16:49:37.754 28307 31219 I ReactNativeJS:   state: 'error' }
```

Note the code (`auth/app-not-authorized`) is just fine, but the typing of Error means you can't get the code, whereas if it was NativeError you may access it.

## Project Files

Here's the line I changed locally (using patch-package) to NativeError, and now things work for me

https://github.com/invertase/react-native-firebase/blob/v5.x.x/src/index.d.ts#L935

My patch-package patch looks like so, I could submit a PR with it if it seems sane enough:

```
diff --git a/node_modules/react-native-firebase/dist/index.d.ts b/node_modules/react-native-firebase/dist/index.d.ts
index f2088b2..19426d1 100644
--- a/node_modules/react-native-firebase/dist/index.d.ts
+++ b/node_modules/react-native-firebase/dist/index.d.ts
@@ -932,7 +932,7 @@ declare module 'react-native-firebase' {
       state: 'sent' | 'timeout' | 'verified' | 'error';
       verificationId: string;
       code: string | null;
-      error: Error | null;
+      error: NativeError | null;
     };
 
     type PhoneAuthError = {
```

I would suppose you'd need a similar one [for the flow type](https://github.com/invertase/react-native-firebase/blob/v5.x.x/src/modules/auth/phone/PhoneAuthListener.js#L20), like so? (note this is a diff local edit of my react-native-firebase fork not a patch-package format diff)

```
diff --git a/src/modules/auth/phone/PhoneAuthListener.js b/src/modules/auth/phone/PhoneAuthListener.js
index e1940601..7edf892f 100644
--- a/src/modules/auth/phone/PhoneAuthListener.js
+++ b/src/modules/auth/phone/PhoneAuthListener.js
@@ -17,7 +17,7 @@ type PhoneAuthSnapshot = {
   state: 'sent' | 'timeout' | 'verified' | 'error',
   verificationId: string,
   code: string | null,
-  error: Error | null,
+  error: NativeError | null,
 };
 
 type PhoneAuthError = {

```

Note that the typing in the callback is PhoneAuthError but the actual object looks like this:

```
'error visible in optional error handler: ', { [Error: This app is not authorized to use Firebase Authentication. Please verifythat the correct package name and SHA-1 are configured in the Firebase Console. [ App validation failed. Is app running on a physical device? ]]
04-08 16:49:37.761 28307 31219 I ReactNativeJS:   code: 'auth/app-not-authorized',
04-08 16:49:37.761 28307 31219 I ReactNativeJS:   nativeErrorMessage: 'This app is not authorized to use Firebase Authentication. Please verifythat the correct package name and SHA-1 are configured in the Firebase Console. [ App validation failed. Is app running on a physical device? ]',
04-08 16:49:37.761 28307 31219 I ReactNativeJS:   line: 119092,
04-08 16:49:37.761 28307 31219 I ReactNativeJS:   column: 26,
04-08 16:49:37.761 28307 31219 I ReactNativeJS:   sourceURL: 'http://10.0.2.2:8081/index.android.delta?platform=android&dev=true&minify=false' }
```

So it looks like a NativeError as well, but that makes me think that the error conversion is incorrect somehow in [src/modules/auth/phone/PhoneAuthListener.js](https://github.com/invertase/react-native-firebase/blob/v5.x.x/src/modules/auth/phone/PhoneAuthListener.js#L292) ?

#### `ios/Podfile`:

- [ ] I'm not using Pods
- [x] I'm using Pods and my Podfile looks like:

```ruby
# N/A
```

#### `AppDelegate.m`:

```objc
// N/A
```

---

#### `android/build.gradle`:

```groovy
// N/A
```

#### `android/app/build.gradle`:

```groovy
// N/A
```

#### `android/settings.gradle`:

```groovy
// N/A
```

#### `MainApplication.java`:

```java
// N/A
```

#### `AndroidManifest.xml`:

```xml
<!-- N/A -->
```

---

## Environment

- **Platform that you're experiencing the issue on**:
  - [ ] iOS
  - [x] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **If known, the version of the platform are you experiencing the issue on:**
  - API25 emulator but any should show it
- **Operating System:**
  - [ ] MacOS, version: `N/A`
  - [ ] Windows, version: `N/A`
  - [ ] Other, please specify: `N/A`
- **Build Tools:**
  - `ADD_SOMETHING_HERE` e.g. Xcode 10, Android Studio 3.2
- **`React Native` version:**
  - `ADD_SOMETHING_HERE`
- **`React Native Firebase` library version:**
  - 5.2.x (about to upgrade to 5.3.x
- **`Firebase` module(s) you're using that has the issue:**
  - [x] **N/A**
  - [x] Authentication
  - [ ] Analytics
  - [ ] Cloud **Firestore**
  - [ ] Cloud **Messaging** (FCM)
  - [ ] Crashlytics
  - [ ] Dynamic **Links**
  - [ ] **Functions** Callable
  - [ ] In App Messaging
  - [ ] Indexing
  - [ ] Invites
  - [ ] Instance ID
  - [ ] ML Kit
  - [ ] Notifications
  - [ ] Performance Monitoring
  - [ ] Realtime **Database**
  - [ ] Remote **Config**
  - [ ] Storage
- **Are you using `TypeScript`?**
  - [ ] No
  - [x] Yes, version: 3.4.x
- **Are you using Expo, e.g. `ExpoKit`?**
  - [x] No
  - [ ] Yes, I've _not_ ejected
  - [ ] Yes, but I **have** ejected to `ExpoKit`
  - [ ] Yes, but I **have** ejected to vanilla React Native
  - Expo version: `N/A`

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Donate via [Open Collective](https://opencollective.com/react-native-firebase/donate)
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter
- 👉 Star this repo on GitHub ⭐️
- 👉 Contribute; see our [contributing guide](./../../CONTRIBUTING.md)
